GoogleSignIn/Sources/GIDAuthentication.m

```diff
@@ -17,8 +17,12 @@
 #import "GoogleSignIn/Sources/GIDAuthentication_Private.h"
 
 #import "GoogleSignIn/Sources/GIDSignInPreferences.h"
+
+#if TARGET_OS_IOS || TARGET_OS_MACCATALYST
```

Comment:
We don't support EMM on Mac Catalyst, so this could be narrowed to iOS.

Reply:
Removed it. How about MDM?

Reply:
The MDM classes are only used by the EMM functionality.

Please apply this change throughout the PR.  EMM support is only needed in the TARGET_OS_IOS case.

Reply:
Done.